openc3/lib/openc3/models/queue_model.rb

```diff
@@ -0,0 +1,152 @@
+# encoding: ascii-8bit
+
+# Copyright 2025 OpenC3, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+require 'openc3/models/model'
+require 'openc3/models/microservice_model'
+require 'openc3/topics/queue_topic'
+require 'openc3/utilities/logger'
+
+module OpenC3
+  class QueueError < StandardError; end
+
+  class QueueModel < Model
+    PRIMARY_KEY = 'openc3__queue'.freeze
+
+    @@class_mutex = Mutex.new
+
+    # NOTE: The following three class methods are used by the ModelController
+    # and are reimplemented to enable various Model class methods to work
+    def self.get(name:, scope:)
+      super("#{scope}__#{PRIMARY_KEY}", name: name)
+    end
+
+    def self.names(scope:)
+      super("#{scope}__#{PRIMARY_KEY}")
+    end
+
+    def self.all(scope:)
+      super("#{scope}__#{PRIMARY_KEY}")
+    end
+    # END NOTE
+
+    def self.queue_command(name, command:, username:, scope:)
+      model = get_model(name: name, scope: scope)
+      raise QueueError, "Queue '#{name}' not found in scope '#{scope}'" unless model
+
+      if model.state != 'DISABLE'
+        Store.rpush("#{scope}:#{name}", { username: username, value: command }.to_json)
+        model.notify(kind: 'command')
+      else
+        raise "Queue '#{name}' is disabled. Command '#{command}' not queued."
```

Comment:
I chose to `raise` if the queue is disabled because a simple log message is easily missed and if you send a command that isn't going to execute I wanted to stop the script.